python/src/scipp/plot/controller.py

```diff
@@ -0,0 +1,468 @@
+            for dim in axes.values():
+
+                coord = self.model.get_data_coord(name, dim)
+
+                if len(coord.dims) > 1:
+                    self.multid_coord = dim
+
+                # To allow for 2D coordinates, the histograms are
+                # stored as dicts, with one key per dimension of the coordinate
+                self.histograms[key][dim] = {}
+                for i, d in enumerate(coord.dims):
+                    self.histograms[key][dim][d] = dim_to_shape[key][
+                        d] == coord_shapes[key][dim][i] - 1
+
+                # The limits for each dimension
+                self.xlims[key][dim] = np.array(
+                    [sc.min(coord).value,
+                     sc.max(coord).value], dtype=np.float)
+                if sc.is_sorted(coord, dim, order='descending'):
+                    self.xlims[key][dim] = np.flip(self.xlims[key][dim]).copy()
+                # Small correction if xmin == xmax
+                if self.xlims[key][dim][0] == self.xlims[key][dim][1]:
+                    if self.xlims[key][dim][0] == 0.0:
+                        self.xlims[key][dim] = [-0.5, 0.5]
+                    else:
+                        dx = 0.5 * abs(self.xlims[key][dim][0])
+                        self.xlims[key][dim][0] -= dx
+                        self.xlims[key][dim][1] += dx
+
+                self.xlims[key][dim] = sc.Variable([dim],
+                                                   values=self.xlims[key][dim],
+                                                   unit=coord.unit)
+
+                self.labels[key][dim] = name_with_unit(var=coord)
+
+        self.initialise_widgets(dim_to_shape[self.name])
+        self.initialise_view(axes)
+        self.initialise_model()
+        if self.profile is not None:
+            self.initialise_profile(axes)
+
+        self.connect_widgets()
+        self.connect_view()
+        if self.panel is not None:
+            self.connect_panel()
+
+    def _ipython_display_(self):
+        return self._to_widget()._ipython_display_()
+
+    def _to_widget(self):
+        return self.widgets._to_widget()
+
+    def initialise_widgets(self, dim_to_shape):
+        parameters = {}
+        for dim in self.labels[self.name]:
+
+            # Dimension labels
+            parameters[dim] = {"labels": self.labels[self.name][dim]}
+
+            # Dimension slider
+            parameters[dim]["slider"] = dim_to_shape[dim]
+
+            # Thickness slider
+            dim_xlims = self.xlims[self.name][dim].values
+            dx = np.abs(dim_xlims[1] - dim_xlims[0])
+            parameters[dim]["thickness_slider"] = dx
+
+            # Slider readouts
+            ind = dim_to_shape[dim] // 2
+            loc = self.model.get_coord_center_value(self.name, dim, ind)
+            parameters[dim]["slider_readout"] = value_to_string(loc)
+
+        self.widgets.initialise(parameters=parameters,
```

Comment:
Suggestion: Aligned with my comment elsewhere. Maybe the initialisation process can be used to direct all the conditional logic relating to the plots that I've noted in the views and SciPlot classes. For example should be able to tell the widgets how to configure themselves based on the model info dimensionality etc.